### ISSUE TITLE
Ingress tls secret is required since v6

### ISSUE THREAD

**Issue by rouke-broersma** [argoproj/argo-helm#2477]
### Describe the bug

I use a custom default wildcard certificate with ingress controller so I don't need to manage multiple certificates in my environments for platform services like argocd. This was supported pre-v6 but with changing the tls field from a list to a bool this is no longer possible. This means I would need to start deploying tls certificates to the argocd namespace where previously I did not need to because the default certificate configured in my ingress controller is valid for argocd. 

### Related helm chart

argo-cd

### Helm chart version

6.0.0

### To Reproduce

Deploy argocd ingress with tls enabled. The ingress will reference a secret that does not exist and the ingress does not become ready. 

### Expected behavior

Ingress controller sees tls is enabled and the secret name is empty thus tries to match with the default certificate. This certificate is valid for the argocd hostname so ingress becomes ready.

**Comment by pdrastil** [comment]
Hi @rouke-broersma you can provide custom TLS configuration with `extraTls` section similar to what was in version 5. The `tls: true` is enabling secret `argocd-server-tls` that Argo CD server expects to provide it's own TLS configuration if using TLS passthrough. Let me know if this helps.

```yaml
extraTls: []
      # - hosts:
      #   - argocd.example.com
      #   secretName: your-certificate-name
```

**Comment by batazor** [comment]
@pdrastil ExtraTls certainly solves the problem, but for such a drastic change it would certainly be a good idea to write a migration guide since many parameters have moved over

**Comment by pdrastil** [comment]
@batazor The info is part of argo-cd README here - https://github.com/argoproj/argo-helm/tree/main/charts/argo-cd#600. I'm happy to improve this docs based on the feedback.

**Comment by batazor** [comment]
@pdrastil Oh thanks, yeah, I usually expected to see the changes in the description of the new version (for example there was a new config ingress) in the end `server.config -> server.cm` had to see for myself how everything changed.

This link solves the problem!

**Comment by pdrastil** [comment]
@batazor After reading your comment I think it might be good idea to create dedicated section in docs with sample setups for various controllers. Whole point behind this change was to decouple aws / gke / generic implementation from single manifest so ingresses can be enhanced without breaking implementations for different controllers.

**Comment by pdrastil** [comment]
@batazor I've enhanced docs with more examples for common setups. Feel free to open PR if something is missing.

Resolved via https://github.com/argoproj/argo-helm/pull/2486

**Comment by rouke-broersma** [comment]
@pdrastil How do I configure the chart for TLS offloading? I see in the diff that my ingress backend port is changing to 443, but I run argocd-server on 80 with server.insecure: 'true' because I do not have TLS in the cluster.

**Comment by pdrastil** [comment]
@rouke-broersma Hi this is good catch as check is against configuration in values.yaml.

Options I see:
- Document that this needs to be set even when params.create: false (probably best option)
- Do lookup for the value, however this might be problematic for people that run chart with GitOps where `helm template` is not performing lookups
- Introduce extra switch as before but it's kinda confusing to have 2 toggles for the same thing.

@mkilchhofer WDYT?

**Comment by mkilchhofer** [comment]
Hard to follow what the question now is, @pdrastil . I cannot follow, sorry🙈. Can you elaborate a bit further?

**Comment by pdrastil** [comment]
@mkilchhofer The question is what to do about mix-in configuration when user generates params by other means that contains `server.insecure: true` and ingress via chart that uses this switch to change ports between 80 / 443.

**Comment by pdrastil** [comment]
@mkilchhofer I've already added the note for given config switch.

```yaml
# -- Run server without TLS
## NOTE: This value should be set when you generate params by other means as it changes ports used by ingress template.
server.insecure: false
```

**Comment by mkilchhofer** [comment]
> @mkilchhofer I've already added the note for given config switch.
> 
> ```yaml
> # -- Run server without TLS
> ## NOTE: This value should be set when you generate params by other means as it changes ports used by ingress template.
> server.insecure: false
> ```

Ah. In my opinion this is enough. 👍 

@rouke-broersma can we close this issue again? Are your points/questions addressed?
